test/common/http/http1/codec_impl_test.cc

```diff
@@ -3317,43 +3255,43 @@ TEST_P(Http1ClientConnectionImplTest, LowWatermarkDuringClose) {
 TEST_P(Http1ServerConnectionImplTest, LargeTrailersRejected) {
   // Default limit of 60 KiB
   std::string long_string = "big: " + std::string(60 * 1024, 'q') + "\r\n\r\n\r\n";
-  testTrailersExceedLimit(long_string, "http/1.1 protocol error: trailers size exceeds limit",
+  testTrailersExceedLimit(long_string, "http/1.1 protocol error: trailers size exceeds limit", true,
                           true);
 }
 
 TEST_P(Http1ServerConnectionImplTest, LargeTrailerFieldRejected) {
   // Construct partial headers with a long field name that exceeds the default limit of 60KiB.
   std::string long_string = "bigfield" + std::string(60 * 1024, 'q');
-  testTrailersExceedLimit(long_string, "http/1.1 protocol error: trailers size exceeds limit",
+  testTrailersExceedLimit(long_string, "http/1.1 protocol error: trailers size exceeds limit", true,
                           true);
 }
 
 // Tests that the default limit for the number of request headers is 100.
 TEST_P(Http1ServerConnectionImplTest, ManyTrailersRejected) {
   // Send a request with 101 headers.
   testTrailersExceedLimit(createHeaderFragment(101) + "\r\n\r\n",
-                          "http/1.1 protocol error: trailers count exceeds limit", true);
+                          "http/1.1 protocol error: trailers count exceeds limit", true, true);
 }
 
 TEST_P(Http1ServerConnectionImplTest, LargeTrailersRejectedIgnored) {
   // Default limit of 60 KiB
   std::string long_string = "big: " + std::string(60 * 1024, 'q') + "\r\n\r\n\r\n";
   testTrailersExceedLimit(long_string, "http/1.1 protocol error: trailers size exceeds limit",
-                          false);
+                          false, /* expect_error = */ parser_impl_ == Http1ParserImpl::BalsaParser);
```

Comment:
The test names on these now with param-conditional expect_error could probably be clarified. Even *without* the test validating different things for different params, the test name is already unclear.